backbone-super/backbone-super.js

```diff
@@ -1,111 +1,128 @@
-		// `parent`'s constructor function.
-		ctor.prototype = parentProto;
-		child.prototype = new ctor();
+      if (_.isObject(source)) {
 
-		// Add prototype properties (instance properties) to the subclass,
-		// if supplied.
-		if (protoProps) {
-			_.extend(child.prototype, protoProps);
+        for (prop in source) {
 
-			// Copy the properties over onto the new prototype
-			for (var name in protoProps) {
-				// Check if we're overwriting an existing function
-				if (typeof protoProps[name] == "function" && fnTest.test(protoProps[name])) {
-					child.prototype[name] = makeWrapper(parentProto, name, protoProps[name]);
-				}
-			}
-		}
+          if (Object.hasOwnProperty.call(source, prop)) {
 
-		// Add static properties to the constructor function, if supplied.
-		if (staticProps) _.extend(child, staticProps);
+            if (_.isFunction(source[prop]) && _.isFunction(obj[prop])) {
+              obj[prop] = makeSuper(source[prop], obj[prop]);
+              obj[prop].prototype = source[prop].prototype;
+            } else {
+              obj[prop] = source[prop];
+            }
+          }
+        }
+      }
+    }
+    return obj;
+  }
+
+  // Backbone's original extend method except swapped out extend
+  // for extend-with-super
+
+  // class properties to be extended.
+  function extend(protoProps, staticProps) {
+    var parent = this;
+    var child;
+
+    // The constructor function for the new subclass is either defined by you
+    // (the "constructor" property in your `extend` definition), or defaulted
+    // by us to simply call the parent's constructor.
+    if (protoProps && _.has(protoProps, 'constructor')) {
+      child = protoProps.constructor;
+    } else {
+      child = function () {
+        return parent.apply(this, arguments);
+      };
+    }
+
+    // Add static properties to the constructor function, if supplied.
+    extendWithSuper(child, parent, staticProps);
 
-		// Correctly set child's `prototype.constructor`.
-		child.prototype.constructor = child;
+    // Set the prototype chain to inherit from `parent`, without calling
+    // `parent`'s constructor function.
+    var Surrogate = function () {
+      this.constructor = child;
+    };
+    Surrogate.prototype = parent.prototype;
+    child.prototype = new Surrogate;
+
+    // Add prototype properties (instance properties) to the subclass,
+    // if supplied.
+    if (protoProps) extendWithSuper(child.prototype, protoProps);
+
+    // Set a convenience property in case the parent's prototype is needed
+    // later.
+    child.__super__ = parent.prototype;
+
+    return child;
+  }
 
-		// Set a convenience property in case the parent's prototype is needed later.
-		child.__super__ = parentProto;
+  // Set up inheritance for the model, collection, router, view and history.
+  Backbone.Model.extend = Backbone.Collection.extend = Backbone.Router.extend = Backbone.View.extend = Backbone.History.extend = extend;
```

Comment:
Remove Backbone.Router as well as Backbone.History as original backbone-super did not modify these namespaces.